mesonbuild/backend/xcodebackend.py

```diff
@@ -248,7 +251,7 @@ def object_filename_from_source(self, target, source):
         project = self.build.project_name
         buildtype = self.buildtype
         tname = target.get_id()
-        arch = platform.machine().lower()
+        arch = self.arch
```

Comment:
I would just skip setting the variable here, and use `{self.arch}` instead of `{arch}` immediately below.

Reply:
Handled in cross compiling commit